docs/source/cuda/simulator.rst

```diff
@@ -0,0 +1,84 @@
+
+.. _simulator:
+
+=================================================
+Debugging CUDA Python with the the CUDA Simulator
+=================================================
+
+Numba includes a CUDA Simulator that implements most of the semantics in CUDA
+Python using the Python interpreter and some additional Python code. This can
+be used to debug CUDA Python code, either by adding print statements to your
+code, or by using the debugger to step through the execution of an individual
+thread.
+
+Execution of kernels is performed by the simulator one block at a time. One
+thread is spawned for each thread in the block, and scheduling of the execution
+of these threads is left up to the operating system.
+
+Using the simulator
+===================
+
+The simulator is enabled by setting the environment variable
+:envvar:`NUMBA_ENABLE_CUDASIM` to 1. CUDA Python code may then be executed as
+normal. The easiest way to use the debugger inside a kernel is to only stop a
+single thread, otherwise the interaction with the debugger is difficult to
+handle. For example, the kernel below will  stop in the thread ``<<<(3,0,0), (1,
+0, 0)>>>``::
+
+    @cuda.jit
+    def vec_add(A, B, out):
+        x = cuda.threadIdx.x
```

Comment:
It could use cuda.grid - whether or not the thread is stopping in `<<<(3,0,0), (1, 0, 0)>>>` then depends on how the kernel was configured. But the example could be changed to demonstrate stopping in a particular thread in the grid, rather than a specific thread/block, if that would seem simpler?